spec/mailers/application_mailer_spec.rb

```diff
@@ -0,0 +1,175 @@
+
+      stub_const('ENV', ENV.to_hash)
+      ENV.delete('MAILGUN_DOMAIN')
+
+      expect(subject.domain).to be_nil
+    end
+
+
+    it 'can be some value (not nil)' do
+      stub_const('ENV', ENV.to_hash.merge('MAILGUN_DOMAIN' => 'blorf.com'))
+      expect(subject.domain).to eq 'blorf.com'
+    end
+
+  end
+
+
+  describe 'content is correct' do
+
+    before(:each) do
+      @test_user = create(:user)
+      @email = ApplicationMailer.test_email(@test_user)
+    end
+
+    it "should be set to be delivered to the email passed in" do
+      expect(@email).to deliver_to(@test_user.email)
+    end
+
+    it "should have the correct subject" do
+      expect(@email).to have_subject( I18n.t('application_mailer.greeting', greeting_name: @test_user.full_name))
+    end
+
+    it "default from address is ENV['SHF_NOREPLY_EMAIL']" do
+      expect(@email).to be_delivered_from(ENV['SHF_NOREPLY_EMAIL'])
+    end
+
+  end
+
+
+  describe 'content is correct for the locale' do
+
+    before(:each) { @orig_local = I18n.locale }
+
+    after(:each) { I18n.locale = @orig_local }
+
+    let(:test_user) { create(:user) }
+
+    it ':en' do
+      I18n.locale = :en
+      email = ApplicationMailer.test_email(test_user)
+      expect(email).to have_subject(I18n.t('application_mailer.greeting', greeting_name: test_user.full_name, locale: :en))
+    end
+
+    it ':sv' do
+      I18n.locale = :sv
+      email = ApplicationMailer.test_email(test_user)
+      expect(email).to have_subject(I18n.t('application_mailer.greeting', greeting_name: test_user.full_name, locale: :sv))
+    end
+
+  end
+
+
+  describe 'use :mailgun delivery method (test mode)' do
+
+    before(:each) { Rails.configuration.action_mailer.delivery_method = :mailgun
+    ApplicationMailer.mailgun_client.enable_test_mode!
+    }
+
+    after(:each) { ApplicationMailer.mailgun_client.disable_test_mode! }
+
+
+    describe 'simple test email (no attachments)' do
+
+      it_behaves_like 'delivery is OK' do
+        let(:email_response) {
+          result = subject.mail(to: 'recipient@example.com', subject: "Test email from SHF", template_name: TEST_TEMPLATE)
+          result.deliver
+        }
+      end
+
+    end
+
+
+    describe 'email with attachments' do
+
+      it_behaves_like 'delivery is OK' do
+        let(:email_response) {
+
+          attachment1 = {
+              content: file_fixture('diploma.pdf'),
+              mime_type: 'application/pdf',
+          }
+          attachment2 = {
+              content: file_fixture('microsoft-word.docx'),
+              mime_type: 'application/docx',
+          }
+
+          result = subject.mail(to: 'recipient@example.com',
+                                subject: 'Test email with attachment',
+                                template_name: TEST_TEMPLATE,
+                                attachments: [attachment1, attachment2])
+          result.deliver
+        }
+      end
+
+    end
```

Comment:
When I ran this test in debug I could not see attachments in the email instance.

I was able to see them with these changes:
```
    describe 'email with attachments' do

      let(:email) do
        subject.attachments['diploma'] = File.read(file_fixture('diploma.pdf'))
        subject.attachments['word_doc'] = File.read(file_fixture('microsoft-word.docx'))

        result = subject.mail(to: 'recipient@example.com',
                              subject: 'Test email with attachment',
                              template_name: TEST_TEMPLATE)

      end

      it_behaves_like 'delivery is OK' do
        let(:email_response) { email.deliver }
      end

      it 'adds attachments to email' do
        expect(email.attachments.length).to eq 2
        expect(email.attachments.first.filename).to eq('diploma')
      end
    end
```
(not that it matters much now since we don't send emails with attachments yet, but I was curious ....)

Reply:
Right -- I haven't implemented attachments yet.

But FYI:  this post was super helpful for me to accomplish this.:
http://www.simonnordberg.com/creating-robust-email-templates-in-action-mailer/

Reply:
in that blog post ^^ he points out that to do attachments, you have to be sure to inline the attachment like this:

```
    def add_inline_attachments!
        attachments.inline['logo.png'] = File.read('/images/logo.png')
    end
```